embed/etcd.go

```diff
@@ -763,6 +763,9 @@ func (e *Etcd) serveMetrics() (err error) {
 
 		for _, murl := range e.cfg.ListenMetricsUrls {
 			tlsInfo := &e.cfg.ClientTLSInfo
+			if !e.cfg.MetricsTLSInfo.Empty() {
+				tlsInfo = &e.cfg.MetricsTLSInfo
```

Comment:
If we override tlsInfo, should we add a warning to log? 

When MetricsTLSInfo is provided, client who has access to other APIs will not be able to access '/metrics' and '/health'?

Reply:
> When MetricsTLSInfo is provided, client who has access to other APIs will not be able to access '/metrics' and '/health'?

Yes. Theoretically, we can make both work (add them to `tls.Config.RootCAs` and to `tls.Config.ClientCAs`), but it will be much more difficult solution because it requires to extend `transport.TLSInfo`.

> If we override tlsInfo, should we add a warning to log?

This is not a hidden behavior (the admin must pass the options explicitly). I am not sure what message do I need to write to log ?
`You specified other certificates for metrics. Access by client certificates will not be possible` ?

Reply:
> This is not a hidden behavior (the admin must pass the options explicitly). I am not sure what message do I need to write to log ?

I was thinking about the scenario where etcd clients with clientTLS (not metricsTLS) might try to access the metrics and health endpoint. Adding something like 'ignoring client certificates since metrics certificates given' would help them understand why the connection is rejected. I am not an expert in this area, please feel free to let me know if you think this is reasonable.

Reply:
@jingyih That's reasonable. I added the message.

Reply:
For better or worse, warning messages are often noted by administrators and automated systems which analyze log output for anomalies. The proposed metrics TLS configuration when used in conjunction with client TLS configuration is a valid state. So in that regard it's not clear to me a preemptive warning makes sense. It seems possible that when using both TLS configurations (again, a valid state), this warning might be interpreted as a false negative alert of some kind.

So, I would suggest we consider changing this to INFO level or omitting it entirely.